test/e2e/e2e_test.go

```diff
@@ -1252,6 +1253,254 @@ var _ = Describe("e2e non-vxlan external gateway and update validation", func()
 	})
 })
 
+// Validate pods can reach a network running in multiple container's loopback
+// addresses via two external gateways running on eth0 of the container without
+// any tunnel encap. This test defines two external gateways and validates ecmp
+// functionality to the container loopbacks. To verify traffic reaches the
+// gateways, tcpdump is running on the external gateways and will exit successfully
+// once an ICMP packet is received from the annotated pod in the k8s cluster.
+var _ = ginkgo.Describe("e2e multiple ecmp external gateway validation", func() {
+	const (
+		svcname             string = "novxlan-externalgw-ecmp"
+		exGWRemoteIpPrefix  string = "10.249.10."
+		ovnNs               string = "ovn-kubernetes"
+		ovnContainer        string = "ovnkube-node"
+		gwContainer1        string = "gw-ecmp-test-container1"
+		gwContainer2        string = "gw-ecmp-test-container2"
+		ovnControlNode      string = "ovn-control-plane"
+		sharedGatewayBridge string = "breth0"
+		ciNetworkName       string = "kind"
+		ecmpRetry           int    = 20
+	)
+
+	f := framework.NewDefaultFramework(svcname)
+
+	type nodeInfo struct {
+		name   string
+		nodeIP string
+	}
+
+	var (
+		workerNodeInfo nodeInfo
+	)
+
+	ginkgo.BeforeEach(func() {
+
+		var (
+			ovnNsFlag                = fmt.Sprintf("--namespace=%s", ovnNs)
+			labelFlag                = fmt.Sprintf("name=%s", ovnContainer)
+			jsonFlag                 = "-o=jsonpath='{.items..metadata.name}'"
+			fieldSelectorControlFlag = fmt.Sprintf("--field-selector=spec.nodeName=%s", ovnControlNode)
+			ovnContainerFlag         = fmt.Sprintf("--container=%s", ovnContainer)
+		)
+
+		// retrieve the ovn-control-plane pod name to determine infra mode to determine if infra is in shared gw mode
+		kubectlOut, err := framework.RunKubectl("get", "pods", ovnNsFlag, "-l", labelFlag, jsonFlag, fieldSelectorControlFlag)
+		if err != nil {
+			framework.Failf("Expected container %s running on %s error %v", ovnContainer, ovnControlNode, err)
+		}
+		ovnPodName := strings.Trim(kubectlOut, "'")
+		// skip the test if the job infra is not running in shared gateway mode by checking if breth0 exists
```

Comment:
this isn't really needed anymore now that local gw and shared gw both use the same topology (aka both have breth0)

Reply:
Excellent, I was digging into some false positive failures from the `framework.skip()` that this will resolve.